scripts/release/other_identifiers/retrieve_indirectIdentifiers_from_mart.pl

```diff
@@ -1,77 +1,156 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl
 use strict;
+use warnings;
 
 use lib "/usr/local/reactomes/Reactome/development/GKB/modules";
 
 use GKB::Config;
-use GKB::DBAdaptor;
-use GKB::EnsEMBLMartUtils qw/:all/;
-use GKB::Utils;
+use GKB::Config_Species;
+use GKB::EnsEMBLMartUtils qw/get_identifiers get_species_mart_name get_wget_query_for_identifier/;
 
-use Data::Dumper;
+use Capture::Tiny qw/capture/;
 use Getopt::Long;
+use List::MoreUtils qw/any/;
 use Try::Tiny;
 
 use Log::Log4perl qw/get_logger/;
-Log::Log4perl->init(\$LOG_CONF);
-my $logger = get_logger(__PACKAGE__);
-
-our($opt_user,$opt_host,$opt_pass,$opt_port,$opt_db,$opt_debug,$opt_sp);
-
-(@ARGV) || die "Usage: $0 -sp 'species name'
--user db_user -host db_host -pass db_pass -port db_port -db db_name\n";
-
-&GetOptions("user:s", "host:s", "pass:s", "port:i", "db:s", "debug", "sp=s");
-
-$opt_db || die "Need database name (-db).\n";    
-
-# Get connection to reactome db
-my $dba = GKB::DBAdaptor->new
-    (
-     -user   => $opt_user,
-     -host   => $opt_host,
-     -pass   => $opt_pass,
-     -port   => $opt_port,
-     -dbname => $opt_db,
-#     -DEBUG => $opt_debug
-     );
-
-# Fetch the species
-$opt_sp ||= 'Homo sapiens';
-
-my $sp = $dba->fetch_instance_by_attribute('Species',[['name',[$opt_sp]]])->[0]
-    || $logger->error_die("No species '$opt_sp' found.\n");
-my $sp_mart_name;
-if ($sp->displayName =~ /^(\w)\w+ (\w+)$/) {
-    $sp_mart_name = lc("$1$2");
-} else {
-    $logger->error_die("Can't form species abbreviation for mart from '" . $sp->displayName . "'.\n");
+
+run(@ARGV) unless caller();
+
+sub run {
+    Log::Log4perl->init(\$LOG_CONF);
+    my $logger = get_logger(__PACKAGE__);
+
+    our($species_code, $help);
+
+    &GetOptions(
+        "sp=s" => \$species_code,
+        "help" => \$help
+    );
+
+    if ($help) {
+        print usage_instructions();
+        exit;
+    }
+
+    foreach my $species_abbreviation (get_species_to_query($species_code)) {
+        my $species_mart_name = get_species_mart_name($species_abbreviation);
+
+        foreach my $identifier (get_identifiers($species_mart_name)) {
+            next if $identifier =~ /chembl|clone_based|dbass|description|ottg|ottt|ottp|shares_cds|merops|mirbase|reactome/;

```

Comment:
Is this worth putting in a config?

Reply:
Not necessary for PR, just a thought

Reply:
Yeah, that's a good idea but I'm kind of inclined to leave it for the moment since other identifiers might be deprecated.  I'll add a TODO comment as a reminder, though.